src/pages/index.tsx

```diff
@@ -46,6 +51,135 @@ export const getServerSideProps: GetServerSideProps<Props, { locale: string }> =
 const HEADER_HEIGHT_S = "107px";
 const HEADER_HEIGHT_M_UP = "96px";
 
+const copyToClipboard = async function (text: string) {
+  try {
+    await navigator.clipboard.writeText(text);
+    console.log("Text copied to clipboard");
+  } catch (error) {
+    console.error("Failed to copy text to clipboard:", error);
+  }
+};
+
+const ShareButton = () => {
+  const { isOpen, open, close } = useDisclosure();
+  const {
+    isOpen: hasInputFocus,
+    open: setFocusOn,
+    close: setFocusOff,
+  } = useDisclosure();
+  const tooltipBoxRef = useRef<HTMLDivElement>(null);
+
+  const linkRef = useRef<HTMLAnchorElement>(null);
+  const mouse = useRef({ x: 0, y: 0 });
+  const handleClick = () => {
+    open();
+    const linkRect = linkRef.current?.getBoundingClientRect() || {
+      x: 0,
+      y: 0,
+      width: 0,
+    };
+    Object.assign(mouse.current, {
+      x: linkRect?.x + linkRect?.width / 2,
```

Comment:
Wondering if there shouldn't be `?? 0` added to both `x` and `width` to prevent NaNs?